Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -1573,7 +1643,39 @@ TEST_F(SerializerTest, DecodesListenResponseWithExistenceFilter) {
   proto.mutable_filter()->set_target_id(100);
 
   SCOPED_TRACE("DecodesListenResponseWithExistenceFilter");
-  ExpectWatchChangeDeserializationRoundTrip(model, proto);
+  ExpectDeserializationRoundTrip(model, proto);
+}
+
+TEST_F(SerializerTest, DecodesVersion) {
+  SnapshotVersion model(Version(123456789).timestamp());
+
+  v1::ListenResponse proto;
+  proto.mutable_target_change()->mutable_read_time()->set_seconds(
+      Version(123456789).timestamp().seconds());
+  proto.mutable_target_change()->mutable_read_time()->set_nanos(
+      Version(123456789).timestamp().nanoseconds());
+
+  SCOPED_TRACE("DecodesVersion");
+  ExpectDeserializationRoundTrip(model, proto);
+}
+
+TEST_F(SerializerTest, DecodesVersionWithNoReadTime) {
+  auto model = SnapshotVersion::None();
+
+  v1::ListenResponse proto;
+
+  SCOPED_TRACE("DecodesVersionWithNoReadTime");
+  ExpectDeserializationRoundTrip(model, proto);
+}
+
+TEST_F(SerializerTest, DecodesVersionWithTargets) {
+  auto model = SnapshotVersion::None();
```

Comment:
Hmm, the test doesn't seem to add a target ID to the model object?

Reply:
Model is `SnapshotVersion`, the point of this test is to verify it will always decode to `None()` if `target_ids` is not empty in the proto.